src/runtime_src/core/pcie/driver/linux/include/xocl_xgq.h

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (C) 2021 Xilinx, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *		 http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+#ifndef _XOCL_XGQ_H_
+#define _XOCL_XGQ_H_
+
+struct xocl_xgq_info {
+	int			 xi_id;
+	u64			 xi_addr;
+	void __iomem		*xi_sq_prod_int;
+};
+
+int xocl_xgq_set_command(void *xgq_handle, int id, u32 *cmd, size_t sz);
+void xocl_xgq_notify(void *xgq_handle);
+int xocl_xgq_get_response(void *xgq_handle, int id);
+int xocl_xgq_attach(void *xgq_handle, void *client, int *client_id);
+
+void *xocl_xgq_init(struct xocl_xgq_info *info);
+void xocl_xgq_fini(void *xgq_handle);
+
+#endif /* _XOCL_XGQ_H_ */
```

Comment:
Now, the name is confusing. there is xocl_xgq and xocl_xgq_cmd in xgq.c. But now we use xocl_xgq for ert.
@maxzhen shall we rename mgmt xgq to:
vmr.c vmr_xgq.c or xgq_vmr.c
and user xgq to
ert.c or ert_xgq.c or xgq_ert.c?

Reply:
when I create xocl_xgq.h and xocl_xgq.c, I was thinking both VMR, ERT can use this library. Let's talk about this offline.

Reply:
> when I create xocl_xgq.h and xocl_xgq.c, I was thinking both VMR, ERT can use this library. Let's talk about this offline.

After chatted with @mamin506, we decided:
1) use xocl_xgq.c as common code, and it will be being improved to accept any common code among ert, vmr and any other potential drivers which utilize xgq.
2) the current xocl_xgq struct in xgq.c should be modified to another name. @xdavidz will file a separate PR after this PR gets merged.

Reply:
> > when I create xocl_xgq.h and xocl_xgq.c, I was thinking both VMR, ERT can use this library. Let's talk about this offline.
> 
> After chatted with @mamin506, we agreed:
> 
>     1. use xocl_xgq.c as common code, and it will be being improved to accept any common code among ert, vmr and any other potential drivers which utilize xgq.
> 
>     2. the current xocl_xgq struct in xgq.c should be modified to another name. @xdavidz will file a separate PR after this PR gets merged.

For 1 above, is xocl_xgq.c a lib which does not maintain its own state, or a subdev which has its own states?

Reply:
> > > when I create xocl_xgq.h and xocl_xgq.c, I was thinking both VMR, ERT can use this library. Let's talk about this offline.
> > 
> > 
> > After chatted with @mamin506, we agreed:
> > ```
> > 1. use xocl_xgq.c as common code, and it will be being improved to accept any common code among ert, vmr and any other potential drivers which utilize xgq.
> > 
> > 2. the current xocl_xgq struct in xgq.c should be modified to another name. @xdavidz will file a separate PR after this PR gets merged.
> > ```
> 
> For 1 above, is xocl_xgq.c a lib which does not maintain its own state, or a subdev which has its own states?

@mamin506 can commit on this. So far, I think only ert is using this comment code and it doesn't maintain its own state. It only provides libs to use the xgq.